packages/app/src/providers/h5wasm/h5wasm-api.ts

```diff
@@ -0,0 +1,153 @@
+import type {
+  ArrayShape,
+  Attribute,
+  AttributeValues,
+  Dataset,
+  Entity,
+  GroupWithChildren,
+  NumericType,
+  Shape,
+} from '@h5web/shared';
+import { hasScalarShape, EntityKind } from '@h5web/shared';
+import type { Dataset as h5wasm_Dataset } from 'h5wasm';
+import {
+  ready as h5wasm_ready,
+  File,
+  Group as h5wasm_Group,
+  FS as h5wasm_FS,
+} from 'h5wasm';
+
+import { ProviderApi } from '../api';
+import type { ExportFormat, ValuesStoreParams } from '../models';
+import { convertDtype, handleAxiosError } from '../utils';
+import type {
+  H5WasmAttribute,
+  H5WasmAttrValuesResponse,
+  H5WasmDataResponse,
+  H5WasmEntityResponse,
+} from './models';
+
+function convert_attrs(attrs: object): Attribute[] {
+  return Object.entries(attrs).map(([name, value]) => ({
+    name,
+    shape: value.shape,
+    type: convertDtype((value as h5wasm_attr).dtype),
+  }));
+}
+
+interface h5wasm_attr {
+  value: unknown;
+  shape: Shape;
+  dtype: string;
+}
+
+export class H5WasmApi extends ProviderApi {
+  /* API compatible with h5wasm@0.2.10 */
+
+  public file_promise: Promise<File>;
+
+  public constructor(filepath: string) {
+    super(filepath);
+    this.file_promise = this.fetchFile(filepath);
+  }
+
+  public async getEntity(path: string): Promise<Entity> {
+    const file = await this.file_promise;
+    const entity_obj = file.get(path);
+    return this.processEntityObject(path, path, entity_obj, true);
+  }
+
+  public async getValue(
+    params: ValuesStoreParams
+  ): Promise<H5WasmDataResponse> {
+    const { dataset } = params;
+    const file = await this.file_promise;
+    const dataset_obj = file.get(dataset.path) as h5wasm_Dataset;
+    const array = dataset_obj.value;
+    return hasScalarShape(dataset) ? array[0] : array;
+  }
+
+  public async getAttrValues(entity: Entity): Promise<AttributeValues> {
+    const { path, attributes } = entity;
+    return attributes.length > 0 ? this.convertAttrValues(path) : {};
+  }
+
+  public getExportURL(
+    dataset: Dataset<ArrayShape, NumericType>,
+    selection: string | undefined,
+    format: ExportFormat
+  ): string | undefined {
+    const { baseURL, params } = this.client.defaults;
+
+    const searchParams = new URLSearchParams(params as Record<string, string>);
+    searchParams.set('path', dataset.path);
+    searchParams.set('format', format);
+
+    if (selection) {
+      searchParams.set('selection', selection);
+    }
+
+    return `${baseURL as string}/data/?${searchParams.toString()}`;
+  }
```

Comment:
Yeah, agreed. If the server provides an endpoint, then it is no longer a simple http server, so serving raw HDF5 files instead of using H5Grove or the like makes a bit less sense.